### DISCUSSION TITLE
How do I set the timeout for the on_close event?

### DISCUSSION THREAD

**Discussion by gbrandt** [flet-dev/flet#924] [Q&A] [answered]
I want to change the timeout for the on_close event from the default 60.  The documentation page says it's a configured timeout but not necessarily a configurable timeout.

Can this timeout be changed within python?

**Comment by ndonkoHenri** [comment]
Yes. There's a variable for that, but I forgot where @FeodorFitsner mentioned the exact name. Maybe `FLET_DEFAULT_TIMEOUT`?

@FeodorFitsner please add a list of all the Flet defaults to the website. Eventually with some brief descriptions. 🙏

**Reply by gbrandt** [reply]
Ok, I will search and see if I can find something like that.  Is it just a matter of overriding the default with my own code" `FLET_DEFAULT_TIMEOUT=xxx`?

I will update this ticket with the exact name if I can find it and if its different than your guess.

**Reply by gbrandt** [reply]
No luck finding a variable that resembles FLET_DEFAULT_TIMEOUT.  No luck finding a variable with a value of 60 or 3600

**Reply by FeodorFitsner** [reply]
Sorry guys, what timeout are you talking about? Could you drop me a link to a place within docs?

**Reply by FeodorFitsner** [reply]
Ah, got it, never mind! Session timeout!

**Reply by FeodorFitsner** [reply]
It's `FLET_ APP_LIFETIME_MINUTES`: https://github.com/flet-dev/flet/blob/main/server/config/config.go#L39

**Reply by gbrandt** [reply]
Ahh, it's in the go code....so this value cannot be changed from python?

**Reply by FeodorFitsner** [reply]
It can be. Add this at the very beginning of your program:

```
os.environ["FLET_APP_LIFETIME_MINUTES"] = "1440"
```

**Reply by ndonkoHenri** [reply]
@FeodorFitsner Please could you document all the env variables flet comes with?

**Comment by gbrandt** [comment, marked as answer]
I can't mark a reply as an answer, so the answer is here. An environment variable FLET_APP_LIFETIME_MINUTES is used for session close

**Reply by ndonkoHenri** [reply]
I got it soooo wrong haha 🤕
